### PR TITLE
TRUNK-5862: Condition and Obs to implement FormRecordable.

### PR DESCRIPTION
**Pull request by icrc-loliveira**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5862

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: api/src/main/java/org/openmrs/BaseFormRecordableOpenmrsData.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import org.apache.commons.lang3.StringUtils;
+import org.openmrs.api.APIException;
+
+import javax.persistence.Column;
+import javax.persistence.MappedSuperclass;
+
+/**
+ * This is a super class to make a bridge between an OpenMrsObject and is position in the form.
+ *
+ * @since 2.3.0
```
Comment (mks-d) [inline api/src/main/java/org/openmrs/BaseFormRecordableOpenmrsData.java:21]: That'd be 2.3.2.

File: api/src/main/java/org/openmrs/FormRecordable.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+/**
+ * Defines the methods to make a bridge between an OpenMrsObject and is position in the form.
+ *
+ * @since  2.3.0
```
Comment (mks-d) [inline api/src/main/java/org/openmrs/FormRecordable.java:15]: Same here :-)

File: api/src/main/java/org/openmrs/Obs.java [hunk 3]
```diff
@@ -1175,93 +1169,15 @@ public void setDateCreated(Date dateCreated) {
...
-	 * @should mark the obs as dirty when the value is changed from a non null to a null value
+	 * @see org.openmrs.FormRecordable#setFormField(String, String)
 	 */
+	@Override
 	public void setFormField(String namespace, String formFieldPath) {
-		if (namespace == null && formFieldPath == null) {
-			markAsDirty(formNamespaceAndPath, null);
-			formNamespaceAndPath = null;
-			return;
-		}
```
Comment (mks-d) [inline api/src/main/java/org/openmrs/Obs.java:1245]: What about this if statement, didn't it get wiped through the changes?

Comment (icrc-loliveira) [reply api/src/main/java/org/openmrs/Obs.java:1245]: This if statement, in on  `BaseFormRecordableOpenmrsData.setFormField` same as version `2.4`

Comment (mks-d) [reply api/src/main/java/org/openmrs/Obs.java:1245]: But not that call:
```java
markAsDirty(formNamespaceAndPath, null);
```
This seems to have been removed through streamlining the code between the base class and `Obs` no?

Comment (icrc-loliveira) [reply api/src/main/java/org/openmrs/Obs.java:1245]: Maybe we can do a discuses this during the meeting tomorrow. 
But markAsDirty will receive the oldValue amd the new value. And I make sure that this still true by saving the oldValue before call the `super.setFormField(namespace, formFieldPath)` and call the `markAsDirty` at the end with the olde value that I saved and the new one calculate in the `setFormField`.
```
private void markAsDirty(Object oldValue, Object newValue) {
```

**Pull request merged**